packages/jest-core/src/lib/viteIntegration.ts

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import type {Config} from '@jest/types';
+
+export type ViteServer = {
+  close: () => Promise<void>;
+  config: unknown;
+};
+
+// Type definition for Vite module (avoiding direct import for optional dependency)
+type ViteModule = {
+  createServer: (config: unknown) => Promise<{
+    close: () => Promise<void>;
+    config: unknown;
+  }>;
+};
+
+/**
+ * Loads Vite module using tsx for TypeScript support
+ */
+async function loadViteModule(): Promise<ViteModule> {
+  try {
+    // Try to load tsx first using dynamic import with string to avoid compile-time dependency
+    const tsxModule = await import(
+      /* @ts-expect-error - Dynamic import of optional peer dependency */
+      /* webpackIgnore: true */ 'tsx/cjs/api'
+    ).catch(() => null);
+
+    if (tsxModule) {
+      const {require: tsxRequire} = tsxModule as {
+        require: (id: string, path: string) => ViteModule;
+      };
+      // Use tsx to load Vite (supports TypeScript configs)
+      const viteModule = tsxRequire('vite', __dirname);
+      return viteModule;
+    }
+
+    // Fall back to regular require if tsx is not available
+
+    return require('vite') as ViteModule;
```

Comment:
This won’t work. Users must use `tsx`. Please remove this fallback. `Vite` is esm so `require` won’t work.

Maybe to avoid optional dependency thing, we should add a local package for Vite integration where Vite and tsx are mandatory deps. Then we use jest core to load that package, similar to `jest-resolve-dependency`